Fix for `c-indentation-indicator`

ingydotnet opened the pull request:
The explicit indentation indicator always meant `1-9`.

This fixes a bug based on a faulty interpretation of the 1.2.1 version.

perlpunk commented on spec/1.2.2/spec.md:
````diff
@@ -4902,7 +4912,7 @@ where detection will fail.
 ```
 [#] c-indentation-indicator ::=
-  ns-dec-digit
+  [x31-x39]             # 1-9
````
The original rule says 
```
[163] | c-indentation-indicator(m) | ::= | ns-dec-digit ⇒ m = ns-dec-digit - #x30
                                           /* Empty */  ⇒ m = auto-detect() |  
```
I don't understand the `ns-dec-digit ⇒ m` part, but on the right site it's already excluding the zero, so maybe this got lost in your conversion?

ingydotnet replied:
In 1.2.1 `m` is the new-indentation parameter.
It needs to be an integer value.

The `ns-dec-digit` production matches a single character `0-9` (should be `1-9`). For example, the character `2` has a hex code point of `x32`. By subtracting x30 you get the integer value.

So `m = ns-dec-digit - #x30` is a (fairly obtuse) way of saying "convert the explicit indentation character to an integer and assign it to `m`".

ingydotnet replied:
Hmmm. That is how I had interpreted it, but you are right.

That is (hopefully) saying "match a ns-dec-digit char except for '0' and assign the (implicitly converted) integer value to `m`.

Sigh. Well at least this one is solved. We still me to fix this as it is currently wrong in `main`. Fixing now...

ingydotnet replied:
`( ns-dec-digit - '0' )` is the clearer way to write that

ingydotnet committed:
This is the correct conversion of this production

iow `m = ns-dec-digit - #x30` means "new indentation is indicated by the
numbers 0-9 except for 0".

Previously thought `- #x30` was a char-to-int operation (which is
implicit now?).

Either way it was awkward way to express it with no textual
clarification.

ingydotnet commented:
Changed from 
```
  ns-dec-digit - x39
```

to the completely straightforward:
```
[x31-x39]    # 1-9
```
